pkg/storage/stores/shipper/indexshipper/tsdb/multitenant.go

```diff
@@ -34,13 +34,14 @@ func withTenantLabelMatcher(userID string, matchers []*labels.Matcher) []*labels
 }
 
 func withoutTenantLabel(ls labels.Labels) labels.Labels {
-	for i, l := range ls {
+	b := labels.NewScratchBuilder(ls.Len())
+	ls.Range(func(l labels.Label) {
 		if l.Name == TenantLabel {
-			ls = append(ls[:i], ls[i+1:]...)
-			break
+			return
 		}
-	}
-	return ls
+		b.Add(l.Name, l.Value)
+	})
```

Comment:
Have you thought about encapsulating this functionality into the `ScratchBuilder` by adding a `Remove(name string)` function?

Reply:
That's a very good point!

Reply:
ScratchBuilder is owned by Promtheus. I'm trying the "normal" builder.